include/aie/Dialect/AIE/IR/AIETargetModel.h

```diff
@@ -232,9 +246,11 @@ class AIETargetModel {
   // Run consistency checks on the target model.
   void validate() const;
 
-  // Return true if this is an NPU-based device
-  // There are several special cases for handling the NPU at the moment.
-  virtual bool isNPU() const { return false; }
+  // Return true if this device has a given property.
+  uint32_t ModelProperties = 0;
```

Comment:
This should be private

Reply:
Good point. Let me fix this in a separate PR.

Reply:
PR created: https://github.com/Xilinx/mlir-aie/pull/1944